libs/blocks/global-navigation/global-navigation.js

```diff
@@ -505,8 +505,49 @@ class Gnav {
     };
 
     const onAnalyticsEvent = (data) => {
-      // TODO: update based on requirements
-      console.log('analytics data', data);
+      if (!data) return;
+
+      const getInteraction = () => {
+        const { event: { type, subtype } = {}, source: { name } = {} } = data;
+        const interactionMapping = {
```

Comment:
Since this is a mostly static Dunamis->AA mapping, I'd move it to the top of the file.

Since the app switcher click mapping has a bit more logic, you could do something such as:
```js
const interactionMapping = {
  ...analyticsMapping,
  'app-switcher': {
    ...analyticsMapping['app-switcher'],
    click: {
      app: getAppLauncherClickAnalyticsName(data)
    }
  }
}
```

If you define a `getAppLauncherClickAnalyticsName` function you could also export it and write a few unit tests for it to make sure the method works in all cases with various invalid input

Reply:
@mokimo - we have a dedicated story for unit testing, the whole implementation would need them.